devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -0,0 +1,124 @@
+/**
+ *  Copyright 2018 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition (name: "ZigBee Accessory Dimmer", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "x.com.st.d.remotecontroller", mnmn: "SmartThings", vid: "generic-dimmer") {
+		capability "Actuator"
+		capability "Switch"
+		capability "Button"
+		capability "Switch Level"
+
+		fingerprint profileId: "0104", inClusters: "0000,0003", outClusters: "0000,0004,0003,0006,0008,0005", manufacturer: "Aurora", model: "Remote50AU", deviceJoinName: "Aurora Wireless Wall Remote"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+		}
+		main "switch"
+		details(["switch"])
+	}
+}
+
+def getSTEP() {10}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		if (event.name=="level" && event.value==0) {}
+		else {
+			sendEvent(event)
+		}
+	} else {
+		def descMap = zigbee.parseDescriptionAsMap(description)
+		if (descMap && descMap.clusterInt == 0x0006) {
+			if (descMap.commandInt == 0x01 || descMap.commandInt == 0x00) {
```

Comment:
Does it use both of these commands? Seems odd if it has one button that can generate two different commands.

Reply:
It does, but I think it's for talking to actual devices that are keeping on/off so it keeps track of state internally

Reply:
I'm wondering if we want to track state separately or just use what the device gives us.  If they have other bulbs that they are controlling directly from the remote it would be weird for us to be generating an "off" event when the bulbs are actually getting an on.

Reply:
I agree, if it's tracking state on it's own I think we should use that.

Reply:
The intended operation with Aurora's hub is to control other lights, I assume.

I don't know any way they'd be able to do this on our system, though. And there's no outward indication on the device or any way for us to determine it's internal state.

Tracking the device's on/off state, but just having a purely invented level value seems counterintuitive to me and also removes the ability to use the app as a "virtual" on/off switch.

Reply:
👍  I see your point